pkg/cmd/pipelinerun/describe_test.go

```diff
@@ -1212,3 +1212,122 @@ func TestPipelineRunDescribeV1beta1_taskrun_with_no_status(t *testing.T) {
+
+	trs := []*v1alpha1.TaskRun{
+		tb.TaskRun("tr-1", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(2*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(5*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-2", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-3", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+	}
+
+	/*
+		pipelinerun.GetAllPipelineRuns( ) returns a slice of strings containing all pipelineruns starting from the most recent one,
+		hence here the recent one is kept as the first element of the slice `pipelineRuns`
+	*/
+	pipelineRuns := []*v1alpha1.PipelineRun{
+		tb.PipelineRun("pipeline-run", "ns",
+			cb.PipelineRunCreationTimestamp(clock.Now()),
+			tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
+			tb.PipelineRunSpec("pipeline"),
+			tb.PipelineRunStatus(
+				tb.PipelineRunTaskRunsStatus("tr-1", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-1",
+					Status:           &trs[0].Status,
+				}),
+				tb.PipelineRunTaskRunsStatus("tr-2", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-2",
+					Status:           &trs[1].Status,
+				}),
+				tb.PipelineRunStatusCondition(apis.Condition{
+					Status: corev1.ConditionTrue,
+					Reason: resources.ReasonSucceeded,
+				}),
+				tb.PipelineRunStartTime(clock.Now()),
+				cb.PipelineRunCompletionTime(clock.Now().Add(15*time.Minute)),
+			),
+		),
+		tb.PipelineRun("pipeline-run2", "ns",
+			cb.PipelineRunCreationTimestamp(clock.Now().Add(5*time.Minute)),
+			tb.PipelineRunLabel("tekton.dev/pipeline", "pipeline"),
+			tb.PipelineRunSpec("pipeline"),
+			tb.PipelineRunStatus(
+				tb.PipelineRunTaskRunsStatus("tr-1", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-1",
+					Status:           &trs[0].Status,
+				}),
+				tb.PipelineRunTaskRunsStatus("tr-3", &v1alpha1.PipelineRunTaskRunStatus{
+					PipelineTaskName: "t-3",
+					Status:           &trs[1].Status,
+				}),
+				tb.PipelineRunStatusCondition(apis.Condition{
+					Status: corev1.ConditionTrue,
+					Reason: resources.ReasonSucceeded,
+				}),
+				tb.PipelineRunStartTime(clock.Now()),
+				cb.PipelineRunCompletionTime(clock.Now().Add(15*time.Minute)),
+			),
+		),
+	}
+
+	namespaces := []*corev1.Namespace{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ns",
+			},
+		},
+	}
+
+	version := "v1alpha1"
```

Comment:
We should use `v1beta1` here as it is the most recent version of the Tekton API.

Reply:
again, v1alpha1 tests rely on test builders (i.e. tb) as you said .. and I am not sure how to write v1beta1 test, still would check it.

Reply:
Example: https://github.com/tektoncd/cli/blob/7f56bc9c76faf67dde19bb878ccc12166c0c67f5/pkg/cmd/pipelinerun/describe_test.go#L917